app/src/main/java/com/niedzielski/volleydemo/MainActivity.java

```diff
@@ -166,11 +168,12 @@ public String toString() {
     private static class Page {
         private int pageid;
         private int ns;
+        private ArrayList<Object> categories;
```

Comment:
It's correct to add the remaining fields you need here but I think it will be easier to work with a more specific class.